influxdb/resource_user.go

```diff
@@ -190,8 +190,8 @@ func updateUser(d *schema.ResourceData, meta interface{}) error {
 
 	if d.HasChange("grant") {
 		oldGrantV, newGrantV := d.GetChange("grant")
-		oldGrant := oldGrantV.([]interface{})
-		newGrant := newGrantV.([]interface{})
+		oldGrant := oldGrantV.(*schema.Set).List()
+		newGrant := newGrantV.(*schema.Set).List()
 
 		for _, oGV := range oldGrant {
```

Comment:
Because these are now `TypeSet`, we can optionally use the `Difference` method:

- https://github.com/hashicorp/terraform/blob/9441e78fb9c35037da71fd0284b97e546dd6a53b/helper/schema/set.go#L109-L120

That could simplify or remove some of this logic. Don't feel required to do so though 😄